test/external-modules/esql-heap-attack/src/javaRestTest/java/org/elasticsearch/xpack/esql/heap_attack/HeapAttackIT.java

```diff
@@ -191,36 +191,53 @@ public void testSortByManyLongsTooMuchMemoryAsync() throws IOException {
         );
     }
 
-    private void assertCircuitBreaks(ThrowingRunnable r) throws IOException {
-        ResponseException e = expectThrows(ResponseException.class, r);
-        Map<?, ?> map = responseAsMap(e.getResponse());
-        logger.info("expected circuit breaker {}", map);
-        assertMap(
-            map,
+    private static final int MAX_ATTEMPTS = 5;
+
+    interface TryCircuitBreaking {
+        Map<String, Object> attempt(int attempt) throws IOException;
+    }
+
+    private void assertCircuitBreaks(TryCircuitBreaking tryBreaking) throws IOException {
+        assertCircuitBreaks(
+            tryBreaking,
             matchesMap().entry("status", 429).entry("error", matchesMap().extraOk().entry("type", "circuit_breaking_exception"))
         );
     }
 
-    private void assertFoldCircuitBreaks(ThrowingRunnable r) throws IOException {
-        ResponseException e = expectThrows(ResponseException.class, r);
-        Map<?, ?> map = responseAsMap(e.getResponse());
-        logger.info("expected fold circuit breaking {}", map);
-        assertMap(
-            map,
+    private void assertFoldCircuitBreaks(TryCircuitBreaking tryBreaking) throws IOException {
+        assertCircuitBreaks(
+            tryBreaking,
             matchesMap().entry("status", 400).entry("error", matchesMap().extraOk().entry("type", "fold_too_much_memory_exception"))
         );
     }
 
+    private void assertCircuitBreaks(TryCircuitBreaking tryBreaking, MapMatcher responseMatcher) throws IOException {
+        int attempt = 1;
+        while (attempt <= MAX_ATTEMPTS) {
+            try {
+                Map<String, Object> response = tryBreaking.attempt(attempt);
+                logger.error("{}: should circuit broken but got {}", attempt, response);
+                attempt++;
+            } catch (ResponseException e) {
+                Map<?, ?> map = responseAsMap(e.getResponse());
+                logger.info("expected circuit breaker {}", map);
```

Comment:
This is expected, I wonder if we should log it at all...

Reply:
Yeah, it really isn't worth it. We get it from the assertion bit if there's a failure.